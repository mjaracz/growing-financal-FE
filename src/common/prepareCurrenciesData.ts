import { CurrenciesResponse } from 'molecules/currencySelect';
import { CurrencyRate } from 'molecules/currencySelect';

export function prepareCurrenciesData(currenciesResponse: CurrenciesResponse): CurrencyRate[] {
  const currenciesKeys: string[] = Object.keys(currenciesResponse.rates);
  const currenciesValues: number[] = Object.values(currenciesResponse.rates);
  const preparedCurrencies = [];
  for (let i = 0; i < currenciesKeys.length; i++) {
    const currencyRate = {}
    currencyRate[currenciesKeys[i]] = currenciesValues[i];
    preparedCurrencies.push(currencyRate)
  }
  return preparedCurrencies;
}