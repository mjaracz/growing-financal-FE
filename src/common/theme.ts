import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: 'hsl(193, 95%, 68%)',
      dark: 'hsl(220, 13%, 18%)'
    },
    secondary: {
      main: 'hsl(281, 100%, 80%)'
    },
    background: {
      default: 'hsl(220, 13%, 18%)'
    },
    text: {
      secondary: 'hsl(0, 0%, 100%)',
      hint: 'rgb(215, 146, 247)'
    }  
  }
})