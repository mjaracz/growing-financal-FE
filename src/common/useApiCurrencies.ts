export const useApiCurrencies = () => {
  const baseUrl = 'http://api.exchangeratesapi.io/v1/latest';
  const apiKey = '481b03536817ae9a3877e3026e237f51';

  const getLatestRate = async () => 
    fetch(baseUrl + `?access_key=${apiKey}`)
      .then(res => res.json())
      .then((data) => {
        if (data.statusCode >= 400) {
          throw new Error('User code error 4** status')
        }
        return data;
      })

  return {
    getLatestRate,
    baseUrl,
    apiKey,
  }
}
