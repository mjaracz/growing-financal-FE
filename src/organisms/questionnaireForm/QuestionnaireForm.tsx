import React from 'react';
import { CustomTextField } from 'atoms/CustomTextField';
import { useStyle } from './useStyle';
import { FormTextBlock } from 'molecules/FormTextBlock/FormTextBlock';
import { CurrencySelect } from 'molecules/currencySelect';
import { useQuestionnaireForm } from './useQuestionnaireForm';
import { CircularProgress } from '@material-ui/core';
import { UploadFile } from 'atoms/UploadFileInput';
import { DatePicker } from 'atoms/DatePicker';
import { Checkbox } from 'atoms/Checkbox';
import { SubmitButton } from 'atoms/SubmitButton';
import { Redirect } from 'react-router-dom'

export const QuestionnaireForm = () => {
  const { 
    status,
    currencies,
    isInvalidName,
    isInvalidEmail,
    validateProjectNameAndSave,
    validateEmailAndSave,
    saveCurrency,
    saveUploadedFile,
    fileName,
    hintUploadText,
    saveDate,
    selectedDate,
    checked,
    saveCheckbox,
    sendFormData,
    hintSubmit,
    redirectToSummary,
  } = useQuestionnaireForm();
  const styles = useStyle();

  if (redirectToSummary) {
    return <Redirect to='/summary' />
  }
  return (
    <div className={styles.questionnaireForm}>
      <div className={styles.formTitle}>
        Questionnaire
      </div>
      <FormTextBlock 
        descContent='First of all feel free to type any name of your dream project.'
        FormTextField={
          <CustomTextField 
            onChange={validateProjectNameAndSave} 
            error={isInvalidName}
            helperText={isInvalidName ? 'Name of project is incorrect or is to short' : null }
          />
        }
      />
      <FormTextBlock
        descContent='Next, enter the contact email.'
        FormTextField={
          <CustomTextField
            onChange={validateEmailAndSave}
            error={isInvalidEmail}
            helperText={isInvalidEmail ? 'Invalid email format (name@example.com)' : null}
          />
        }
      />
      <FormTextBlock
          descContent='Type a currency which is most convenient for you.'
          FormTextField={(status === 'loading') 
            ? <CircularProgress size={80} color='primary' /> 
            : <CurrencySelect currenciesOptions={currencies} handleInputChange={saveCurrency} />
          }
      />
      <FormTextBlock 
        descContent='If you have any layout design feel free to upload it to us. Allowed file format, jpg, png. '
        FormTextField={
          <UploadFile 
            fileName={fileName} 
            handleUploadFile={saveUploadedFile} 
            hintText={hintUploadText} 
          />
        }
      />
      <FormTextBlock
        descContent='When you plan to start a project ?'
        FormTextField={
          <DatePicker 
            selectedDate={selectedDate}
            saveDate={saveDate}
          />
        }
      />
      <FormTextBlock
        descContent='One last thing, please check if you have no criminal record and are legally allowed to be in the U.S.'
        FormTextField={
          <Checkbox 
            checked={checked} 
            handleCheckboxChange={saveCheckbox} 
          />
        }
      />
      <div className={styles.submitWrapper}>
        <SubmitButton 
          sendFormData={sendFormData} 
          label='submit'
        />
        <p className={styles.submitHint}>
          {hintSubmit}
        </p>
      </div>
    </div>
  )
}