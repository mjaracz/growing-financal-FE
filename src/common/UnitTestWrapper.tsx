import React, { FC } from 'react';
import { ThemeProvider } from '@material-ui/core';
import { theme } from './theme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'redux/store';

export const UnitTestWrapper: FC = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    </BrowserRouter>
  )
}