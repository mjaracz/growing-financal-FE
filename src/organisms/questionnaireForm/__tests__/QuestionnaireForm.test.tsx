import React from 'react';
import { render } from '@testing-library/react';
import { QuestionnaireForm } from '../QuestionnaireForm';
import { UnitTestWrapper } from 'common/UnitTestWrapper';

describe('<QuestionnaireForm />', () => {
  it('should match the snapshot', () => {
    const { baseElement } = render(
      <UnitTestWrapper>
        <QuestionnaireForm />
      </UnitTestWrapper>
    ) 
    expect(baseElement).toMatchSnapshot()
  });
})