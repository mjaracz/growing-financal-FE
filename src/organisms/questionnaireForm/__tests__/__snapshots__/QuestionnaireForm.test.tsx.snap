// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionnaireForm /> should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-questionnaireForm-1"
    >
      <div
        class="makeStyles-formTitle-2"
      >
        Questionnaire
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          First of all feel free to type any name of your dream project.
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-8"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-10"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          Next, enter the contact email.
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-8"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-10"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          Type a currency which is most convenient for you.
        </p>
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 80px; height: 80px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          If you have any layout design feel free to upload it to us. Allowed file format, jpg, png. 
        </p>
        <div
          class="makeStyles-uploadWrapper-13"
        >
          <input
            class="makeStyles-uploadFileInput-14"
            id="file"
            type="file"
          />
          <label
            class="makeStyles-uploadLabel-15"
            for="file"
          >
            Select file
          </label>
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          When you plan to start a project ?
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-8 MuiFormControl-marginNormal"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
          >
            Choose date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              type="text"
              value="29/06/2021"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                aria-label="change date"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
              >
                <span>
                  Choose date
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          Choose a primary color tone for your new project
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-8"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-10"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formTextBlok-6"
      >
        <p
          class="makeStyles-textBlockDescription-7"
        >
          One last thing, please check if you have no criminal record and are legally allowed to be in the U.S.
        </p>
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorPrimary makeStyles-textField-8 MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-20"
              data-indeterminate="false"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
      </div>
      <div
        class="makeStyles-submitWrapper-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-21 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <p
          class="makeStyles-submitHint-5"
        />
      </div>
    </div>
  </div>
</body>
`;
